File: internal/sast/fixtures/csharp/Xxe_positive.cs
// internal/sast/fixtures/csharp/Xxe_positive.cs
using System.Xml;
using Microsoft.AspNetCore.Mvc;

public class XxeController : Controller
{
    public IActionResult Parse(string xml)
    {
        var doc = new XmlDocument();
        doc.LoadXml(xml);
        return Content(doc.DocumentElement.Name);
    }
}

File: internal/sast/fixtures/javascript/xxe_positive.js
// internal/sast/fixtures/javascript/xxe_positive.js
const libxmljs = require("libxmljs");
const express = require("express");
const app = express();
app.use(express.text({ type: "application/xml" }));

app.post("/parse", (req, res) => {
  const doc = libxmljs.parseXmlString(req.body, { noent: true });
  res.json({ root: doc.root().name() });
});
